tools/pylint-extensions/pylint-guidelines-checker/pylint_guidelines_checker.py

```diff
@@ -888,22 +908,39 @@ def visit_return(self, node):
             if node.parent.parent.name.endswith("Client") and node.parent.parent.name not in self.ignore_clients and node.parent.is_method():
                 if node.parent.name.startswith("list"):
                     paging_class = False
+                    async_itempaged_return = False
 
                     try:
-                        if any(v for v in node.value.infer() if "def by_page" in v.as_string()):
-                            paging_class = True
-                    except (astroid.exceptions.InferenceError, AttributeError, TypeError): # astroid can't always infer the return
+                        for value in node.value.infer():
+                            if str(value) == "Uninferable":
+                                logger.debug("Pylint custom checker failed to check if client list method uses core paging.")
+                                return
+                            elif value.name == "ItemPaged" and isinstance(node.parent, astroid.FunctionDef):
+                                paging_class = True
+                            elif value.name == "AsyncItemPaged" and isinstance(node.parent, astroid.AsyncFunctionDef):
```

Comment:
I'm not sure an async paging method will be captured as an astroid.AsyncFunctionDef since it's not defined with `async def`